CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc

```diff
@@ -1668,20 +1778,280 @@ void EcalTPGParamBuilder::analyze(const edm::Event& evt, const edm::EventSetup&
     for (int ich = 0; ich < (int)my_StripEcalLogicId2_EE.size(); ich++) {
       FEConfigWeightDat wut;
       int igroup = 1;  // this group is for EE
+      weights_map_even[stripMapEE[my_StripEcalLogicId2_EE[ich].getLogicID()]] = 1;
       wut.setWeightGroupId(igroup);
       // Fill the dataset
       dataset2[my_StripEcalLogicId2_EE[ich]] = wut;
     }
 
-    // Insert the datasets
-    ostringstream wtag;
-    wtag.str("");
-    wtag << "Shape_NGroups_" << NWEIGROUPS;
-    std::string weight_tag = wtag.str();
-    ss << " weight tag " << weight_tag << "\n";
-    if (m_write_wei == 1)
-      wei_conf_id_ = db_->writeToConfDB_TPGWeight(dataset, dataset2, NWEIGROUPS, weight_tag);
+    if (writeToDB_) {
+      // Insert the datasets
+      ostringstream wtag;
+      wtag.str("");
+      wtag << "Shape_NGroups_" << NWEIGROUPS;
+      std::string weight_tag = wtag.str();
+      ss << " weight tag " << weight_tag << "\n";
+      if (m_write_wei == 1)
+        wei_conf_id_ = db_->writeToConfDB_TPGWeight(dataset, dataset2, NWEIGROUPS, weight_tag);
+    }
+    /////////////////////////////////////////////////////////
+  } else {
+    // Read Even weights from file.
+    std::vector<std::vector<unsigned int>> weights_even;
+
+    std::ifstream file(weight_even_weightGroupFile_);
+    std::string line;
+    while (std::getline(file, line)) {
+      std::istringstream iss(line);
+      std::vector<unsigned int> wei_group_even;
+      for (std::string s; iss >> s;)
+        wei_group_even.push_back(stoi(s));
+      weights_even.push_back(wei_group_even);
+    }
+    int nweigroups_even = weights_even.size();
+
+    std::ifstream file2(weight_even_idMapFile_);
+    std::string line2;
+    while (std::getline(file2, line2)) {
+      std::istringstream iss(line2);
+      int ch_id, wei_gr;
+      iss >> std::dec >> ch_id;
+      iss >> std::dec >> wei_gr;
+      int wei_group = wei_gr;
+      if (wei_group > nweigroups_even - 1) {
+        edm::LogError("TopInfo") << "Associating strip: " << ch_id << " to non-existing weight group: " << wei_gr
+                                 << "\n";
+      }
+      weights_map_even[ch_id] = wei_group;
+    }
+
+    map<EcalLogicID, FEConfigWeightGroupDat> dataset_even;
+
+    for (int igrp = 0; igrp < nweigroups_even; igrp++) {
+      if (weights_even[igrp].size() == 5) {
```

Comment:
Is this "5" a fundamental parameter? Perhaps, it can be taken from somewhere else, where it is defined. Or at least define a named constant for it, as it returns in several other places in this code

Reply:
converted to a constexpr defined in the header